admin/authentication/local-strategy.js

```diff
@@ -41,19 +41,21 @@ module.exports = async function (req, email, password, done) {
       return done(null, false)
     }
 
-    const mock = {
+    const sessionData = {
       EmailAddress: email,
       UserName: email,
       UserType: 'SchoolNom',
       School: school.dfeNumber,
       role: role.title,
-      logonAt: Date.now()
+      logonAt: Date.now(),
+      userId: user.id
     }
 
     // Success - valid login
     logonEvent.user_id = user.id
-    await saveValidLogonEvent(logonEvent, mock)
-    return done(null, mock)
+    req.session.userId = user.id
```

Comment:
Seems odd that this is required here.  Am I missing something?

Reply:
this is basically adding the userId to the session object, so it can be used to pass to services that need to record it (such as the settingService).

Reply:
It's already added in to the sessionData above - I don't think this line `req.session.userId = user.id` is needed.

Reply:
my discovery in manual testing suggests otherwise.  Maybe we have an issue here?
i'll do some further testing with the other values

Reply:
found it! its under `req.session.passport.user`